File: pset2/readability.c
#include<cs50.h>
#include<stdio.h>
#include<string.h>
#include<math.h>


int main(void)
{
    //get string
    string s = get_string("Text: ");

    //declare variables
    int espaces = 0;
    int letters = 0;
    int words = 0;
    int sent = 0;

    //go to every character and feed the variables correctly
    for(int i=0, n=strlen(s); i < n; i++)
    {
        if((s[i] >= 'a' && s[i] <= 'z') || (s[i] >= 'A' && s[i] <= 'Z'))
        {
            letters += 1;
        }

        if (s[i] == 32)
        {
            espaces += 1;
        }

        if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            sent += 1;
        }
    }

    //determine the number of words
    words = espaces + 1;

    float L = (letters * 100) / words;
    float S = (100 * sent) / words;
    float ind = 0.0588 * L - 0.296 * S - 15.8;
    int index = round(ind);

    //print results
    if (ind < 2)
    {
        printf("Before Grade 1\n");
    }
    if (ind > 16)
    {
        printf("Grade 16+\n");
    }
    if (ind>2 && ind<16)
    {
        printf("Grade %i\n", index);
    }
}

File: pset6/readability.py
s = input("Text: ")

spaces = 0
letters = 0
word = 1
sent = 0

for i in s:
    if i == " ":
        word = word + 1
        spaces = spaces + 1
    if i =="." or i == "!" or i =="?":
        sent = sent + 1

letters = len(s) - spaces - sent

L = (letters * 100)/word
S = (100 * sent)/word
ind = 0.0588*L - 0.296*S - 15.8

index = int(ind)

if ind < 2:
    print("Before Grade 1")
if ind > 16:
    print("Grade 16+")

if ind>2 and ind<16:
    print(f"Grade {index}")

